res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -1712,6 +1940,10 @@ PioneerDDJSX.samplerLedsPlay = function(value, group, control) {
         sampleDeck = "[Sampler" + (samplerIndex + 1) + "]",
         padNum = PioneerDDJSX.samplerGroups[sampleDeck];
 
+    if (!engine.getValue(sampleDeck, "duration")) {
```

Comment:
Sometimes the `track_loaded` control behaved different than using `duration` and confused me. I could change the if-clause to request if the `duration` is equal to 0, would that be better?